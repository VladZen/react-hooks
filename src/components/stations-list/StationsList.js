import '@styles/components/stations-list.scss'
import Station from '@components/stations-list/StationsListItem'

// temp
import stations from '../../mock.json'

const StationsList = () => {
  return (
    <ul className='stations'>
      {stations.map((station) => (
        <Station
          key={ station.station_ID }
          data={ station }
        />
      ))}
    </ul>
  )
}

export default StationsList
