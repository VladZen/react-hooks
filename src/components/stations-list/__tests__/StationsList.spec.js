import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import StationsList from '@/components/stations-list/StationsList'
import MockStationsListItem from '@/components/stations-list/StationsListItem'

jest.mock('@/components/stations-list/StationsListItem', () => {
  return function Mock (props) {
    return (
      <li data-testid="list-item">
        { props.station.name }
      </li>
    )
  }
})

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

const mockStations = [
  {
    'station_ID': 102,
    'custom_evse_id': null,
    'location_ID': 102,
    'seller_ID': 27,
    'name': 'Endesa 2',
    'connected': 1,
    'position': '60.868623,26.702901',
    'available': 1,
    'lastconnect': '2019-02-22T13:36:12.000Z',
    'roaming_identifier_cpo': 'FI*001'
  },
  {
    'station_ID': 103,
    'custom_evse_id': null,
    'location_ID': 103,
    'seller_ID': 28,
    'name': 'Endesa 2',
    'connected': 1,
    'position': '60.868623,26.702901',
    'available': 0,
    'lastconnect': '2019-02-22T13:36:12.000Z',
    'roaming_identifier_cpo': 'FI*001'
  }
]

describe('StationListItem component', () => {
  describe('rendering', () => {
    it('with items', () => {
      act(() => {
        render(
          (
            <StationsList stations={ mockStations } />
          ),
          container
        )
      })

      expect(container).toMatchSnapshot()
    })

    it('empty list', () => {
      act(() => {
        render(
          (
            <StationsList stations={ null } />
          ),
          container
        )
      })

      expect(container).toMatchSnapshot()
    })
  })
})
