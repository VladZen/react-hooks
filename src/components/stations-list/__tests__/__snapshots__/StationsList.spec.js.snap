// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationListItem component rendering empty list 1`] = `
<div>
  <ul
    class="stations"
  >
    <li
      class="text-center"
    >
      There are no stations in the list
    </li>
  </ul>
</div>
`;

exports[`StationListItem component rendering with items 1`] = `
<div>
  <ul
    class="stations"
  >
    <li
      data-testid="list-item"
    >
      Endesa 2
    </li>
    <li
      data-testid="list-item"
    >
      Endesa 2
    </li>
  </ul>
</div>
`;
